### PR TITLE
feature: Add pagination or load more for blog posts in - posts & series

### LINKED ISSUE #9: Add pagination or load more for blog posts in - `posts` & `series`
**Additional context**
Right now the existing blog query is retrieving the first 10 blog posts, this is the initial state to get things going.
A pagination needs to be implemented that when the user navigates to the next page a different query is used to retrieve the next 10 posts and the next and so on.

In the query this is done by specifying `after 10`  I believe.

### PR DESCRIPTION
**Pull request by jhonacs2**
## This PR Closes Issue 
closes #9 

## Description
Add pagination or load more for blog posts in - posts & series

## What type of PR is this? (check all applicable)

- [x] 🅰️ Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added to documentation?
- [ ] 📜 README.md
- [x] 🙅 no documentation needed

### CODE REVIEW

File: angular-app/src/app/components/posts/posts.component.ts [hunk 1]
```diff
@@ -1,29 +1,53 @@
...
 import { AsyncPipe } from '@angular/common';
 import { Post } from '../../models/post';
+import { PageInfo } from '../../models/blog-info';
+import { InfiniteScrollDirective } from '../../directives/infinite-scroll.directive';
 
 @Component({
   selector: 'app-posts',
   standalone: true,
-  imports: [RouterLink, AsyncPipe],
+  imports: [RouterLink, AsyncPipe, InfiniteScrollDirective],
```
Comment (monacodelisa) [inline angular-app/src/app/components/posts/posts.component.ts:12]: @jhonacs2 very nice implementation, I like custom directives, nicely done

File: angular-app/src/app/graphql.operations.ts [hunk 2]
```diff
@@ -47,31 +47,36 @@ query Publication ($host: String!) {
...
-    posts(first: 10) {
-      edges {
-        node {
-          id,
-          slug,
-          coverImage {
-            url
+export const GET_POSTS=
+   gql`
+    query Publication ($host: String!, $first: Int!, $after: String!){
```
Comment (monacodelisa) [inline angular-app/src/app/graphql.operations.ts:52]: @jhonacs2 the only thing I am thinking about is - do we need the `after` number to be a `variable` . Is it going to vary, or are we always going to be incrementing with `10` 🤔But that seems like more of a future question that we can figure out later on. For now this seems like a great solution 👍

File: angular-app/src/app/services/blog.service.ts [hunk 3]
```diff
@@ -46,13 +53,24 @@ export class BlogService {
     .valueChanges.pipe(map(({ data }) => data.publication.author));
   }
 
-  getPosts(host: string): Observable<Post[]> {
+  getPosts(host: string, first: number = 10, after: string = ''): Observable<BlogPaginationInfo> {
```
Comment (monacodelisa) [inline angular-app/src/app/services/blog.service.ts:56]: @jhonacs2 if `first` is an `number` but it is always `10` maybe `after` can be just `first to string` or something like that. I am just writing things down so I can revisit them later on, for now you do not need to do any changes looks good going to merge.

**Review by monacodelisa — APPROVED** [review]

**Review by copypasteov — APPROVED** [review]

**Review by cleancodecraft — APPROVED** [review]

**Comment by monacodelisa** [comment]
@jhonacs2 the `load more` doesn't show up for me, will have to see what is going on

**Comment by monacodelisa** [comment]
@jhonacs2 I looked it up by `after` they meant a cursor position 😉 not after as in a certain amount of posts

**Pull request merged**